### DISCUSSION TITLE
Help needed for app_commands.choice max choices

### DISCUSSION THREAD

**Discussion by NavisGames** [Rapptz/discord.py#9241] [Q&A] [answered]
Hello! I have a few app_command options for some roulette stuff, now roulette lets you select a single number from 1 to 36 instead of a "group of numbers".

Now you only have 25 choices, but I have many more choices

This command is inspired by UnbelieaveaBoat, which offers well over 25 choices.

How to add it in discord.py too?

```py
@tree.command(
    name="roulette", description="Play roulette and multipliers high multipliers!"
)
@app_commands.describe(
    number="Amount of money in your wallet that you want to gamble with! (e.g. 10 or all)"
)
@app_commands.choices(
    fields=[
        app_commands.Choice(name="🔴 Red", value="RED"),
        app_commands.Choice(name="⚫ Black", value="BLACK"),
        app_commands.Choice(name="🔶 even", value="EVEN"),
        app_commands.Choice(name="🔷 odd", value="ODD"),
        app_commands.Choice(name=" 0", value="GREEN"),
        app_commands.Choice(name="1-12", value="1ST"),
        app_commands.Choice(name="13-24", value="2ND"),
        app_commands.Choice(name="25-36", value="3RD"),
        app_commands.Choice(name="1st column", value="2TO1 1"),
        app_commands.Choice(name="2nd column", value="2TO1 2"),
        app_commands.Choice(name="3rd column", value="2TO1 3"),
    ]
)
async def roulette(
    Interaction: Discord.Interaction,
    number: st,
    fields: app_commands.Choice[str],
):
#...
```

Here is some of the code needed.

**Comment by ika2kki** [comment, marked as answer]
Autocomplete options and choices have a fixed limit at 25, can't go past that.

Discord API does however support optional minimum and maximum limits for numeric and string types. So to pick a number between 1-36, you can annotate to a [`Range`](https://discordpy.readthedocs.io/en/stable/interactions/api.html#discord.app_commands.Range), for example:
```py
@app_commands.command()
async def command(interaction: discord.Interaction, number: app_commands.Range[int, 1, 36]):
    ...
```

**Reply by NavisGames** [reply]
okay imma try it out thanks!

**Reply by NavisGames** [reply]
i tried it out, it would work but i want to make the 1 tro 36 numbers and the other choices that are in the code ^^

**Reply by ika2kki** [reply]
> i tried it out, it would work but i want to make the 1 tro 36 numbers and the other choices that are in the code ^^

Won't be able to make a choice for each number between 1-36 since you can only have up to 25 per argument.

I'd recommend keeping 1-36 as a range type, and categorising each of the other options into their own arguments
For example, red and black can be a `colour` argument, even or odd can be `parity` and 1st/2nd/3rd columns can be a `column` argument.

**Reply by NavisGames** [reply]
okay. Weird that a Bot from a User named Unbelieveaboat can have more than 25 choices
![image](https://user-images.githubusercontent.com/83245135/219646165-f641a04d-491f-4151-a300-426885dd0773.png)
![image](https://user-images.githubusercontent.com/83245135/219646191-1aec88da-e8c1-41a2-8a3f-4994f0108161.png)
![image](https://user-images.githubusercontent.com/83245135/219646241-de8b905d-8e4c-4723-96b9-4db323ac3813.png)

**Reply by LeoCx1000** [reply]
What is returned by autocomplete is dynamic, not fixed. In your last screenshot you started typing something hence the options changed dynamically. You can return whatever you want from an [autocomplete callback](https://discordpy.readthedocs.io/en/stable/interactions/api.html#discord.app_commands.autocomplete) so it's technically unlimited, but the limit is still 25 choices displayed <ins>at once.</ins>

Hope this clarifies it. 🙂
